### ISSUE TITLE
Hide inherited fields / methods / operators

### ISSUE THREAD

**Issue by ArtemKolichenkov** [dart-lang/dartdoc#2057]
I am aware of #641 and #980. 
PR #1324 says #641 is fixed, but it didn't address the issue of hiding inherited things.

## The problem

Take a look at [average pub.dev API reference](https://pub.dev/documentation/search_widget/latest/search_widget/MySingleChoiceSearchState-class.html).  
The amount of noise is just annoying. After all this is **API reference**, most of inherited things are not related to package/widget API at all.  
From end-user perspective who wants to use a package those autogenerated inherited things like `toDiagnosticsNode` or `hashCode` don't help one bit to understand **this particular package** and I'm pretty sure 99.9% of people just skim through it thinking "nah its some unimportant internal dart thing". It feels very out of place.  

Sometimes it gets even more ridiculous like with the following thing that is present in almost every dart package:
```
operator ==(dynamic other) → bool
The equality operator.
inherited
```
I mean - do we really need to tell developers how `==` operator works in documentation of every library ever published? Sometimes somebody needs to read docs about `==`, but I'm pretty sure this person would just go and find the docs for it specifically and not while lurking though docs for some fancy sparkling widget.

I think this is quite a huge problem especially for newcomers.  
Class might implement 2 "new" things and only those 2 things are of any interest to readers, but instead docs bombard them with tens of inherited things. I might navigate through it easily, but I've been reading various docs every day for the last 10 years. Newcomer might just have migraine trying to filter what's important.

## Possible solutions

There is two sides of the coin to this I guess:
###  As a package maintainer
I'd like to have options to:
- Disable generating inherited stuff when running `dartdoc` locally, both to make it faster and to not see useless inherited things while I'm documenting
- Have the ability to tell pub.dev to exclude inherited things from generated API reference (if not possible - read "As documentation reader")
### As documentation reader
I'd like to have the option to hide (not dim!) things that are not related to actual package API reference. Of course if something is heavily overridden and deserves attention - keep it, but there is no point showing me `==` operator just because it was used in the code once.

**Comment by jcollins-g** [comment]
I agree some more flexible rendering options would be nice here; people have a wide variety of preference to the level of detail they like to see in docs.

**Comment by jcollins-g** [comment]
@srp7474 Is it the goal for locally generated documentation (outside of the pub site, Flutter documentation, etc) to have this feature?  Or were you envisioning something more ecosystem wide?

Also, it sounds like what is wanted here could be a flag to:

- Hide _all_ inherited fields/methods/operators
- Hide inherited fields/methods/operators defined outside the package being documented (that way, inheritance within a package would still show up, but Dart SDK methods like `hashCode` or things from other packages like `args`, `path`, or Flutter, etc. would not).
- Hide inherited fields/methods/operators defined in the Dart SDK.

The implementation difference between the three is actually fairly small -- once we have a way to hide things selecting which ones we hide based on some criteria is pretty easy.

The main reason this is P3 though is not implementation path, but concern about the larger ecosystem and expectations, particularly if we begin allowing packages to define this themselves on pub.

**Comment by srp7474** [comment]
@jcollins-g Yes I intend to publish them as it is a framework that is a methodology for state manipulation using a mathematical model.

I understand the dilemma about not breaking the universe apart for a seemingly innocuous change.

All I really need is to hide inherited fields from the top level (subclass) class. They are my own classes that I am extending and I cannot document the subclass without a ton of unnecessary stuff showing up that is already defined in the documentation for the class being extended.

Steve

**Comment by ses110** [comment]
Is there progress or a visible date for implementing this feature? If not, are there any workarounds I can use to hide methods inherited from classes outside of my project?

Thanks

**Comment by srawlins** [comment]
There is no progress or estimated date. I don't know of any workarounds...

**Comment by mstepanov214** [comment]
possible quick-and-dirty solution:

`header.html`
```html
 <style type="text/css">
   .inherited {
     display: none !important;
   }
 </style>
```
`dartfoc-options.yaml`
```yaml
dartdoc:
  header: ['doc/header.html']
```

**Comment by srawlins** [comment]
Just a little mock-up of an idea of _muted_ inherited members. If we did not show all of the members of List which are inherited from Iterable, it would lead to too much confusion. We could mute with the option to expand:

**Before**

<img width="748" alt="Screen Shot 2022-08-01 at 9 43 54 AM" src="https://user-images.githubusercontent.com/103167/182202090-421ddea4-a77f-4acb-8572-cab33108b434.png">

**After**
<img width="558" alt="Screen Shot 2022-08-01 at 9 50 35 AM" src="https://user-images.githubusercontent.com/103167/182202125-7145108f-8989-4f63-8a9e-aed739e63dcb.png">
